Atomic/storage.py

```diff
@@ -38,6 +41,29 @@ def reset(self):
         except:
             selinux.restorecon(root)
 
+    def modify(self):
+        try:
+            shutil.copyfile(self.dss_conf, self.dss_conf_bak)
+            if len(self.args.devices) > 0:
+                self._add_device(self.args.devices)
+            if self.args.driver:
+                self._driver(self.args.driver)
+            if util.call(["docker-storage-setup"]) != 0:
+                os.rename(self.dss_conf_bak, self.dss_conf)
+                util.call(["docker-storage-setup"])
+                raise ValueError("Not all devices could be added")
+        finally:
+            if os.path.exists(self.dss_conf_bak):
+                os.remove(self.dss_conf_bak)
+
+    def _add_device(self, devices):
+        util.sh_modify_var_in_file(self.dss_conf, "DEVS",
+                                   lambda old: util.sh_set_add(old, devices))
+
+    def _driver(self, driver):
+        util.sh_modify_var_in_file(self.dss_conf, "STORAGE_DRIVER",
+                                   lambda old: util.sh_set_add(old, [driver]))
```

Comment:
`STORAGE_DRIVER` is not a set, so the modification function should be simply `lambda old: driver`.